experimental/trusted_shuffler/client/src/main.rs

```diff
@@ -32,27 +35,106 @@ pub struct Opt {
         default_value = "http://localhost:8080"
     )]
     server_url: String,
+    #[structopt(long, help = "The QPS we are aiming to simulate", default_value = "10")]
+    qps: u32,
+    #[structopt(
+        long,
+        help = "How many rounds, i.e., seconds, we sent requests",
+        default_value = "1"
+    )]
+    rounds: u32,
 }
 
-#[tokio::main]
+#[tokio::main(flavor = "multi_thread", worker_threads = 2)]
 async fn main() -> anyhow::Result<()> {
-    env_logger::init();
+    env_logger::builder()
+        .format_timestamp(None)
+        .format_level(false)
+        .format_module_path(false)
+        .format_target(false)
+        .init();
     let opt = Opt::parse();
 
-    let request = "Request 1";
-    info!("Sending request: {}", request);
-    let url = format!("{}/request", &opt.server_url);
-
-    let request_start = Instant::now();
-    let response = send_request(&url, Method::POST, request.as_bytes())
-        .await
-        .context("Couldn't receive response")?;
-    let request_duration = request_start.elapsed();
-
-    let parsed_response = String::from_utf8(response).context("Couldn't decode response body")?;
-    info!(
-        "Received response: {} in {:?}",
-        parsed_response, request_duration
-    );
+    let expected_qps = opt.qps;
+    let rounds = opt.rounds;
+
+    // Start with giving one second iterations.
+    let start_time = Instant::now();
+
+    // Header of table printed by logger.
+    log::info!("phase,id,request_elapsed_in_ms,response_elapsed_in_ms");
+
+    let mut clients = vec![];
```

Comment:
AFAICT these are not really clients, just futures that terminate once each round trip is done, right?

Reply:
I think the functionality is fine, it's just that the name is not very accurate IMO.

Reply:
I added a comment clarifying that a client sends only one request and terminates.